tests/Sulu/Component/Content/Mapper/ContentMapperTest.php

```diff
@@ -3187,6 +3187,122 @@ public function testContentTypeSwitch()
         $this->mapper->save($testSiteData, 'internal-link', 'sulu_io', 'en', 1, true, $uuid);
     }
 
+    /**
+     * It should delete a node which has children with history.
+     * It should not throw an exception
+     */
+    public function testDeleteWithChildrenHistory()
+    {
+        $data = array(
+            array(
+                'title' => 'A',
+                'url' => '/a',
+            ),
+            array(
+                'title' => 'B',
+                'url' => '/a/b',
+            ),
+            array(
+                'title' => 'C',
+                'url' => '/a/b/c',
+            ),
+            array(
+                'title' => 'D',
+                'url' => '/a/d',
+            ),
+        );
+
+        // save content
+        $data[0] = $this->mapper->save($data[0], 'overview', 'sulu_io', 'de', 1);
+        $data[1] = $this->mapper->save($data[1], 'overview', 'sulu_io', 'de', 1, true, null, $data[0]->getUuid());
+        $data[2] = $this->mapper->save($data[2], 'overview', 'sulu_io', 'de', 1, true, null, $data[1]->getUuid());
+        $data[3] = $this->mapper->save($data[3], 'overview', 'sulu_io', 'de', 1, true, null, $data[0]->getUuid());
+
+        // move /a/b to /a/d/b
+        $this->mapper->move($data[1]->getUuid(), $data[3]->getUuid(), 1, 'sulu_io', 'de');
+
+        // delete /a/d
+        $this->mapper->delete($data[3]->getUuid(), 'sulu_io');
+
+        // check
+        try {
+            $this->mapper->load($data[3]->getUuid(), 'sulu_io', 'de');
+            $this->fail('Node should not exist');
```

Comment:
@wachterjohannes fyi, I changed this to only check that the node we asked to be deleted is deleted. if that was deleted, then its children for sure were deleted also :) (or else we have a very big jackalope problem)..